cli/pcluster/pcluster.py

```diff
@@ -691,6 +691,29 @@ def _get_param_value(params, key_name):
     return next((i.get("ParameterValue") for i in params if i.get("ParameterKey") == key_name), None)
 
 
+def _get_ip(stack, outputs, config):
+    """
+    Get public / private ip for a cluster.
+
+    :param stack: stack name
+    :param outputs: stack outputs
+    :param config: config object
+    :return: ip else None
+    """
+    # Get Master Public IP, if use_public_ips = true
+    ip = _get_output_value(outputs, "MasterPublicIP")
+    if ip:
+        return ip
+
+    # get master ip even if use_public_ips = false but vpc creates a public ip
+    ip = _get_master_server_ip(stack, config)
+    if ip:
+        return ip
+
+    # fall back to private ip for users that can resolve a private ip locally
+    return _get_output_value(outputs, "MasterPrivateIP")
```

Comment:
I think `_get_master_server_ip` is already trying to get private IP and exiting if unsuccessful. 

This function could be reduced to:
```
return _get_output_value(outputs, "MasterPublicIP") or _get_master_server_ip(stack, config)
```

(probably we could drop the function completely)

Reply:
yep, you're right